### PR TITLE
Fix CI packaging

### PR DESCRIPTION
**Pull request by TimeEngineer**
# What has changed ?

The CI packaging is broken, because the `default-feature` isn't disabled for `libparsec_platform_device_loader` which depends on `libparsec_crypto`

- [ ] Does this PR affect the User ?

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -124,6 +124,13 @@ jobs:
         env:
           POETRY_LIBPARSEC_BUILD_STRATEGY: no_build
 
+      - name: Additional Check
+        run: |
+          set -ex
+          rustup target add wasm32-unknown-unknown
+          cargo check --no-default-features --features use-sodiumoxide
+          cargo check --target wasm32-unknown-unknown
```
Comment (FirelightFlagboy) [inline .github/workflows/ci.yml:132]: cargo check could use the build artifact to be faster.
So move it to `test-rust-matrix` like for the task `pre-commit clippy`

**Review by FirelightFlagboy — COMMENTED** [review]
Another way would be to remove completely the `features` that aren't compatible between desktop and web from `default-features` :thinking:

**Review by Volham22 — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (TimeEngineer) [reply .github/workflows/ci.yml]: Is it better ?

File: build.py [hunk 2]
```diff
@@ -64,14 +64,11 @@ def build():
     # native module and discard the rest !
 
     maturin_build_profile = "--profile=" + os.environ.get("CARGO_PROFILE", DEFAULT_CARGO_PROFILE)
-    features = (
-        "--no-default-features --features use-sodiumoxide"
-        if maturin_build_profile == "--profile=release"
-        else ""
-    )
+    maturin_build_features = os.environ.get("CARGO_FEATURES", "")
```
Comment (touilleMan) [inline build.py:67]: This environ variable can be use to inject any parameter, hence we should rename it as `CARGO_EXTRA_ARGS`

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -373,6 +373,14 @@ jobs:
         shell: bash
         run: echo OPENSSL_SRC_PERL=C:/Strawberry/perl/bin/perl >> $GITHUB_ENV
 
+      - name: Additional Check
```
Comment (touilleMan) [inline .github/workflows/ci.yml:376]: This check is quiet costly: we re-compile the entire project and dependencies (and we add pressure on the cache)

Instead what we should do:
- modify `.github/actions/paths-filter` to add a new `rust-platform` entry this match `oxidation/libparsec/crates/platform_*/**`[^1]
- add this `rust-platform` to the `client-web` filter

Now `🌐 Web tests` will be triggered any time the platform code is modified ;-)

[^1]: we can also be a bit more cunning in filtering and have instead a `rust-platform-web` that match on `oxidation/libparsec/crates/platform_*/src/web/**`, this way we avoid triggering the web build if changing the native part

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -373,6 +373,14 @@ jobs:
...
+        if: startsWith(matrix.os, 'ubuntu')
+        run: |
+          set -ex
+          rustup target add wasm32-unknown-unknown
+          cargo check --no-default-features --features use-sodiumoxide
```
Comment (touilleMan) [inline .github/workflows/ci.yml:381]: What the need for that ? I would say check is basically the first half of compilation, so this line is redundant with when we do `cargo test --profile ci-rust --package libparsec_crypto --no-default-features --features use-sodiumoxide`

**Review by touilleMan — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 4 above)
Comment (TimeEngineer) [reply .github/workflows/ci.yml:381]: Here we compile all with sodiumoxide, that's how we can detect if the package will work

File: build.py (hunk 2 above)
Comment (TimeEngineer) [reply build.py:67]: Done

File: .github/workflows/ci.yml (hunk 1 above)
Comment (FirelightFlagboy) [reply .github/workflows/ci.yml]: Yes because we could use the cache

Comment (FirelightFlagboy) [reply .github/workflows/ci.yml]: But the important part is to measure

File: .github/workflows/ci.yml (hunk 3 above)
Comment (TimeEngineer) [reply .github/workflows/ci.yml]: Done

**Review by FirelightFlagboy — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -373,6 +373,10 @@ jobs:
         shell: bash
         run: echo OPENSSL_SRC_PERL=C:/Strawberry/perl/bin/perl >> $GITHUB_ENV
 
+      - name: Additional Check (for packaging)
+        if: startsWith(matrix.os, 'ubuntu')
+        run: cargo check --no-default-features --features use-sodiumoxide
```
Comment (touilleMan) [inline .github/workflows/ci.yml:378]: add `--profile ci-rust`

File: build.py [hunk 6]
```diff
@@ -64,14 +64,11 @@ def build():
     # native module and discard the rest !
 
     maturin_build_profile = "--profile=" + os.environ.get("CARGO_PROFILE", DEFAULT_CARGO_PROFILE)
-    features = (
-        "--no-default-features --features use-sodiumoxide"
-        if maturin_build_profile == "--profile=release"
-        else ""
-    )
+    maturin_build_features = os.environ.get("CARGO_EXTRA_ARGS", "")
```
Comment (touilleMan) [inline build.py:67]: rename variable to match env var

**Review by touilleMan — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (TimeEngineer) [reply .github/workflows/ci.yml]: It takes 3mins to check but we win 2mins in rust test (so 1min remains)
And it was approximately ~ 2mins

**Pull request merged**